apps/shared/avifjpeg.c

```diff
@@ -1064,7 +1064,9 @@ static avifBool avifJPEGReadInternal(FILE * f,
                 for (size_t c = 0; c < AVIF_JPEG_EXTENDED_XMP_GUID_LENGTH; ++c) {
                     // According to Adobe XMP Specification Part 3 section 1.1.3.1:
                     //   "128-bit GUID stored as a 32-byte ASCII hex string, capital A-F, no null termination"
-                    if (((guid[c] < '0') || (guid[c] > '9')) && ((guid[c] < 'A') || (guid[c] > 'F'))) {
+                    // Also allow lowercase since some cameras use lowercase. https://github.com/AOMediaCodec/libavif/issues/2755
+                    if (((guid[c] < '0') || (guid[c] > '9')) && ((guid[c] < 'A') || (guid[c] > 'F')) &&
+                        ((guid[c] < 'a') || (guid[c] > 'f'))) {
```

Comment:
Optional: We can now use the [`isxdigit`](https://en.cppreference.com/w/c/string/byte/isxdigit) function declared in <ctype.h>:

```
                    if (!isxdigit(guid[c])) {
```